scripts/ci/jobs/diff-dumps.sh

```diff
@@ -121,13 +121,19 @@ upload_offline_dump() {
     latest_version=$(grep -oE '^[0-9]+\.[0-9]+' "$version_file" | sort -V | tail -n 1)
 
     file_to_check="scanner-v4-defs-${latest_version}.zip"
+    temp_file="/tmp/$file_to_check"
 
-    if curl --silent --show-error --max-time 60 --retry 3 -o $file_to_check https://definitions.stackrox.io/v4/offline-bundles/$file_to_check; then
-        # If the file exists, add it to scanner-vuln-updates.zip
-        zip scanner-vuln-updates.zip "$file_to_check"
-        echo "$file_to_check added to scanner-vuln-updates.zip"
+    if curl --silent --show-error --max-time 60 --retry 3 -o "$temp_file" "https://definitions.stackrox.io/v4/offline-bundles/$file_to_check"; then
+        if file "$temp_file" | grep -q 'Zip archive data'; then
```

Comment:
Do you know why `curl` was successful, but the output file was invalid/incorrect/bogus?

Reply:
I am guessing probably `curl --silent --show-error --max-time 60 --retry 3 -o $file_to_check` is saving error messages as `$file_to_check`

maybe `if curl --silent --show-error --max-time 60 --retry 3 -o "$temp_file" "https://definitions.stackrox.io/v4/offline-bundles/$file_to_check"` doesn't check status code?

Reply:
Yeah, your guesses and maybes are possible, you probably want to look at `man curl`.

Reply:
I decided to use `jq .` and it should return parse error if there's no json file exists.

Reply:
You're missing `--fail`:

```
       -f, --fail
              (HTTP) Fail fast with no output at all on server errors. This  is
              useful to enable scripts and users to better deal with failed at‐
              tempts.  In  normal  cases when an HTTP server fails to deliver a
              document, it returns an HTML document  stating  so  (which  often
              also  describes  why and more). This command line option prevents
              curl from outputting that and return error 22.
```

With that and `set -e`, non-successful HTTP requests will make `curl` return non-zero, and the shell will exit.

I still think it's worth doing the `jq` sanity check to ensure it's a valid JSON. It's a good sanity check. We should avoid ingesting invalid input into the pipeline.

Reply:
Short-sight on my side: The file is a zip archive, so using `jq` for sanity check doesn't make sense.

Maybe we can use `unzip -l`. That will show us the zip file's content.

Reply:
Yeah, I was confused that we should check the json file.